pandasql/io.py

```diff
@@ -17,3 +40,52 @@ def read_numpy(*args, name=None, **kwargs):
 
 def read_pickle(*args, name=None, **kwargs):
     return DataFrame(pd.read_pickle(*args, **kwargs), name=name)
+
+
+def _csv_to_sql(*args, name, **kwargs):
+    kwargs['nrows'] = SAMPLE_LINES
+    chunk = pd.read_csv(*args, **kwargs)
+    chunk.to_sql(name=name, con=SQL_CON, index=False, if_exists='append')
+    subprocess.call(["sqlite3", DB_FILE, ".mode csv",
+                     f".import  \'| tail -n +{SAMPLE_LINES + 2} {args[0]}\' {name}"])   # noqa
+
+    df = pd.DataFrame(columns=chunk.columns)
+    return DataFrame(df, name=name, offload=False, loaded_on_sqlite=True)
+
+
+def _csv_chunking(*args, name, **kwargs):
+    kwargs['chunksize'] = CHUNKSIZE
+    kwargs['nrows'] = None
+    for chunk in pd.read_csv(*args, **kwargs):
+        chunk.to_sql(name=name, con=SQL_CON, index=False, if_exists='append')
+        cols = chunk.columns
+
+    df = pd.DataFrame(columns=cols)
+    return DataFrame(df, name=name, offload=False, loaded_on_sqlite=True)
```

Comment:
Same as above, don't pass in a df since that will be cached incorrectly on Pandas. Set columns explicitly.